src/libraries/Microsoft.PowerFx.LanguageServerProtocol/LanguageServer/LanguageServer.cs

```diff
@@ -300,22 +301,24 @@ private void HandleCompletionRequest(string id, string paramsJson)
             // Avoid this if possible
             _logger?.Invoke($"[PFX] HandleCompletionRequest: Suggest results: Count:{result.Suggestions.Count()}, Suggestions:{string.Join(", ", result.Suggestions.Select(s => $@"[{s.Kind}]: '{s.DisplayText.Text}'"))}");
 
-            var items = new List<CompletionItem>();
-
-            foreach (var item in result.Suggestions)
+            var precedingCharacter = expression[cursorPosition - 1];
+            _sendToClient(JsonRpcHelper.CreateSuccessResult(id, new
             {
-                items.Add(new CompletionItem()
+                items = result.Suggestions.Select((item, index) => new CompletionItem()
                 {
                     Label = item.DisplayText.Text,
                     Detail = item.FunctionParameterDescription,
                     Documentation = item.Definition,
-                    Kind = GetCompletionItemKind(item.Kind)
-                });
-            }
+                    Kind = GetCompletionItemKind(item.Kind),
 
-            _sendToClient(JsonRpcHelper.CreateSuccessResult(id, new
-            {
-                items,
+                    // The order of the results should be preserved.  To do that, we embed the index
+                    // into the sort text, which clients may sort lexigraphically.
+                    SortText = index.ToString(CultureInfo.InvariantCulture),
```

Comment:
>SortText [](http://example.com/codeflow?start=20&length=8)

This may intersect with the work, [https://msazure.visualstudio.com/OneAgile/_workitems/edit/24935881](https://msazure.visualstudio.com/OneAgile/_workitems/edit/24935881)
Is this change for the same?

Reply:
Yes it is. We are coordinating on this. Jack was gracious enough to take care of this :)

Reply:
Navaneeth is taking it to finish though :)

Reply:
Yes , my initial work fixed it in the  UFB side (thought was the approach decided). But definitely this is a better and the right way to fix(on the server side). We would need to make changes in UFB to respect this (server side) completion result and filtering/sorting that comes with it.